pkg/apis/softwarecomposition/types.go

```diff
@@ -596,3 +597,104 @@ type SBOMSyftFilteredList struct {
 
 	Items []SBOMSyftFiltered
 }
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type Framework struct {
+	metav1.TypeMeta
+	metav1.ObjectMeta
+	ControlsRef []FrameworkControlRef
+	Payload     unstructured.Unstructured
+}
+
+type FrameworkControlRef struct {
+	Name       string
+	ControlID  string
+	Kind       string
+	ApiVersion string
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type FrameworkList struct {
+	metav1.TypeMeta
+	metav1.ListMeta
+	Items []Framework
+}
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type Control struct {
+	metav1.TypeMeta
+	metav1.ObjectMeta
+	RulesRef []ControlRuleRef
+	Payload  unstructured.Unstructured
+}
+
+type ControlRuleRef struct {
+	Name       string
+	Kind       string
+	ApiVersion string
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type ControlList struct {
+	metav1.TypeMeta
+	metav1.ListMeta
+	Items []Control
+}
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type Rule struct {
+	metav1.TypeMeta
+	metav1.ObjectMeta
+	Rego    string
+	Payload unstructured.Unstructured
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type RuleList struct {
+	metav1.TypeMeta
+	metav1.ListMeta
+	Items []Rule
+}
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type Exception struct {
+	metav1.TypeMeta
+	metav1.ObjectMeta
+	Payload unstructured.Unstructured
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type ExceptionList struct {
+	metav1.TypeMeta
+	metav1.ListMeta
+	Items []Exception
+}
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type ControlConfiguration struct {
+	metav1.TypeMeta
+	metav1.ObjectMeta
+	Payload unstructured.Unstructured
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type ControlConfigurationList struct {
+	metav1.TypeMeta
+	metav1.ListMeta
+	Items []ControlConfiguration
+}
```

Comment:
chore: fix line endings in PR

Please see the sign after this line. It indicates that the files have improper line endings: they don’t have the terminating `\n` characters. I believe it’s because you’re using Visual Studio Code or some other IDE that terminates the files with Windows line endings.

Please refer to this issue: https://github.com/microsoft/vscode/issues/141169

Reply:
I'll fix this,

also the fact that VSC does this is just concerning, I'd at least hope to have my tooling follow standards :(

Reply:
Yeah, that’s a common problem. Nonetheless, we could always add a Git pre-commit hook to fix the line endings automatically or a linter to complain about it. But as you can imagine, this is very low on our priorities list :)